analysis/mage/src/StandardChecks.tsx

```diff
@@ -0,0 +1,161 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import CASTS_THAT_ARENT_CASTS from 'parser/core/CASTS_THAT_ARENT_CASTS';
+import { SpellInfo } from 'parser/core/EventFilter';
+import {
+  EventType,
+  AnyEvent,
+  CastEvent,
+  BeginChannelEvent,
+  BeginCastEvent,
+} from 'parser/core/Events';
+import Events from 'parser/core/Events';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import EventHistory from 'parser/shared/modules/EventHistory';
+
+class StandardChecks extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    eventHistory: EventHistory,
+  };
+  protected abilityTracker!: AbilityTracker;
+  protected eventHistory!: EventHistory;
+
+  castBreakdownDuringBuff(buff: SpellInfo) {
+    /**
+     * @param buff the spell object for the buff
+     * @returns an array containing each unique spell cast and the number of times it was cast
+     */
+    const castEvents: CastEvent[] = this.getEventsDuringBuff(buff, EventType.Cast);
+    const castArray: number[][] = [];
+    castEvents &&
+      castEvents.forEach((c) => {
+        const index = castArray.findIndex((arr) => arr.includes(c.ability.guid));
+        if (index !== -1) {
+          castArray[index][1] += 1;
+        } else {
+          castArray.push([c.ability.guid, 1]);
+        }
+      });
+    return castArray;
```

Comment:
I really don't like this 2D array return type. The 2nd dimension of the array is being used like an unlabeled object where the 0 index is implicitly the spellId and the 1 index is implicitly count. This should explicitly be an object instead. This will have the additional effect of making the downstream code more readable:
```
const castArray: { spellId: number, count: number }[] = [];
    castEvents &&
      castEvents.forEach((c) => {
        const index = castArray.findIndex((arr) => arr.spellId === c.ability.guid);
        if (index !== -1) {
          castArray[index].count += 1;
        } else {
          castArray.push({ spellId: c.ability.guid, count: 0 });
        }
      });
```